File: shared/src/commonMain/kotlin/com/conservatio/domain/model/Project.kt
package com.conservatio.domain.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class Project(
    val id: String,
    val title: String,
    val clientId: String?,
    val objectIds: List<String>,
    val status: ProjectStatus,
    val startDate: Instant?,
    val endDate: Instant?,
    val description: String?,
    val totalBudget: Double?,
    val currency: String?,
    val createdAt: Instant,
    val updatedAt: Instant,
    val syncStatus: SyncStatus
)

@Serializable
enum class ProjectStatus {
    INQUIRY, QUOTED, APPROVED, IN_PROGRESS, ON_HOLD, COMPLETED, ARCHIVED
}

File: iosApp/Conservatio/Models/Project.swift
import Foundation

enum ProjectStatus: String, CaseIterable, Codable {
    case inquiry = "Inquiry"
    case quoted = "Quoted"
    case approved = "Approved"
    case inProgress = "In Progress"
    case onHold = "On Hold"
    case completed = "Completed"
    case archived = "Archived"

    var icon: String {
        switch self {
        case .inquiry: return "questionmark.circle"
        case .quoted: return "doc.text"
        case .approved: return "checkmark.circle"
        case .inProgress: return "hammer"
        case .onHold: return "pause.circle"
        case .completed: return "checkmark.seal"
        case .archived: return "archivebox"
        }
    }
}

struct Project: Identifiable, Codable {
    var id: String = UUID().uuidString
    var title: String = ""
    var clientName: String = ""
    var status: ProjectStatus = .inquiry
    var objectIds: [String] = []
    var startDate: Date?
    var endDate: Date?
    var budget: Double?
    var currency: String = "EUR"
    var description: String = ""
    var createdAt: Date = Date()
}
